actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -637,6 +637,12 @@ def mount(app, options = nil)
             options = app
             app, path = options.find { |k, _| k.respond_to?(:call) }
             options.delete(app) if app
+
+            raise ArgumentError, "A rack application must be specified" unless app.respond_to?(:call)
```

Comment:
This seems like an unrelated change?

Reply:
I wanted to make the exception raise without a deprecation warning if you pass a hash without an app. I can try to think of another way that's more obvious, but maybe that's actually ok.